angular-workspace/projects/ni/nimble-angular/src/thirdparty/directives/radio_control_value_accessor.ts

```diff
@@ -0,0 +1,225 @@
+ *
+ * @ngModule ReactiveFormsModule
+ * @ngModule FormsModule
+ * @publicApi
+ */
+@Directive({
+  selector:
+      'input[type=radio][formControlName],input[type=radio][formControl],input[type=radio][ngModel]',
+  host: {'(change)': 'onChange()', '(blur)': 'onTouched()'},
+  providers: [RADIO_VALUE_ACCESSOR]
+})
+export class RadioControlValueAccessor extends BuiltInControlValueAccessor implements
+    ControlValueAccessor, OnDestroy, OnInit {
+  /** @internal */
+  // TODO(issue/24571): remove '!'.
+  _state!: boolean;
+  /** @internal */
+  // TODO(issue/24571): remove '!'.
+  _control!: NgControl;
+  /** @internal */
+  // TODO(issue/24571): remove '!'.
+  _fn!: Function;
+
+  /**
+   * The registered callback function called when a change event occurs on the input element.
+   * Note: we declare `onChange` here (also used as host listener) as a function with no arguments
+   * to override the `onChange` function (which expects 1 argument) in the parent
+   * `BaseControlValueAccessor` class.
+   * @nodoc
+   */
+  override onChange = () => {};
+
+  /**
+   * @description
+   * Tracks the name of the radio input element.
+   */
+  // TODO(issue/24571): remove '!'.
+  @Input() name!: string;
+
+  /**
+   * @description
+   * Tracks the name of the `FormControl` bound to the directive. The name corresponds
+   * to a key in the parent `FormGroup` or `FormArray`.
+   */
+  // TODO(issue/24571): remove '!'.
+  @Input() formControlName!: string;
+
+  /**
+   * @description
+   * Tracks the value of the radio input element
+   */
+  @Input() value: any;
+
+  constructor(
+      renderer: Renderer2, elementRef: ElementRef, private _registry: RadioControlRegistry,
+      private _injector: Injector) {
+    super(renderer, elementRef);
+  }
+
+  /** @nodoc */
+  ngOnInit(): void {
+    this._control = this._injector.get(NgControl);
+    this._checkName();
+    this._registry.add(this._control, this);
+  }
+
+  /** @nodoc */
+  ngOnDestroy(): void {
+    this._registry.remove(this);
+  }
+
+  /**
+   * Sets the "checked" property value on the radio input element.
+   * @nodoc
+   */
+  writeValue(value: any): void {
+    this._state = value === this.value;
+    this.setProperty('checked', this._state);
+  }
+
+  /**
+   * Registers a function called when the control value changes.
+   * @nodoc
+   */
+  override registerOnChange(fn: (_: any) => {}): void {
+    this._fn = fn;
+    this.onChange = () => {
+      fn(this.value);
+      this._registry.select(this);
+    };
+  }
+
+  /**
+   * Sets the "value" on the radio input element and unchecks it.
+   *
+   * @param value
+   */
+  fireUncheck(value: any): void {
+    this.writeValue(value);
+  }
+
+  private _checkName(): void {
+    /* [Nimble] Commenting because ngDevMode is not defined
+    if (this.name && this.formControlName && this.name !== this.formControlName &&
+        (typeof ngDevMode === 'undefined' || ngDevMode)) {
+      throwNameError();
+    }
```

Comment:
It may be a long day, but the comment seems to indicate that ngDevMode is always the value undefined, then `(typeof ngDevMode === 'undefined' || ngDevMode)` should always resolve to true right?

That seems like different behavior than what this patch is doing. Here it is behaving as if ngDevMode is always the value `false` instead of `undefined`.

Are we sure ngDevMode is always undefined? If so then it seems like normally this expression would run. I think we expect this check to run as it notifies the developer that the formControlName and radio name are not the same.

Reply:
Uncommented this code, as well as throwNameError(). Version 14.2.6 of the file throws a RuntimeError which takes an additional enum parameter. That enum is not defined the same way in version 12, and so I'm reverting this one part of the file back to the version 12 behavior, which is to throw Error instead of RuntimeError. This should be a trivial difference which we can leave in even after we've updated to Angular 14. Or if we remember to, we can restore version 14.2.6 of that line.

Reply:
You can add a note to the Angular 13, 14 upgrade issue that captures the change needed to align with Angular 14 error behavior https://github.com/ni/nimble/issues/270